edweiss-app/components/core/PeriodBlock.tsx

```diff
@@ -3,32 +3,61 @@ import TView from './containers/TView';
+ */
+export const PeriodBlock = ({
+    period,
+    course,
+    user,
+    format
+}: {
+    period: CourseTimePeriod;
+    course: { id: string; data: Course; };
+    user: any;
+    format: string;
 }) => {
-    const direction = format == 'week' ? 'column' : 'row';
-    const size1 = format == 'week' ? 12 : 15;
-    const size2 = format == 'week' ? 9 : 12;
+    // Set layout direction and text sizes based on the format ('day' or 'week')
+    const direction = format === 'week' ? 'column' : 'row';
+    const textSizePrimary = format === 'week' ? 12 : 15;
+    const textSizeSecondary = format === 'week' ? 9 : 12;
+
     return (
-        <TView flexDirection="column">
+        <TView flexDirection="column" testID='period-block-view'>
             <TView justifyContent="space-between" flexDirection={direction}>
-                <TText color="constantBlack" numberOfLines={1} size={size1} p={5}>
+                <TText color="constantBlack" numberOfLines={1} size={textSizePrimary} p={5}>
                     {`${period.type.charAt(0).toUpperCase() + period.type.slice(1)}`}
                 </TText>
-                <TText pl={5} pr={10} pt={7} color="overlay2" numberOfLines={1} size={size2}>
-                    {`${period.rooms.join(", ")}`}
+                <TText pl={5} pr={10} pt={7} color="overlay2" numberOfLines={1} size={textSizeSecondary}>
+                    {period.rooms.join(", ")}
                 </TText>
             </TView>
-            <TText pl={5} pb={5} pt={3} color="overlay0" numberOfLines={1} size={size2}>
-                {`${course.data.name}`}
+
+            <TText pl={5} pb={5} pt={3} color="overlay0" numberOfLines={1} size={textSizeSecondary}>
+                {course?.data.name}
             </TText>
+
             <TView flexDirection={direction}>
-                <TText p={3} size={size2} color="overlay2">
+                <TText p={3} size={textSizeSecondary} color="overlay2">
                     {`${formatTime(period.start)} - ${formatTime(period.end)}`}
                 </TText>
             </TView>
-            {user.data.type == 'student' && course.data.started && period.type == 'lecture' && <TText p={5} size={15} color="red">Join Course</TText>}
-            {user.data.type == 'professor' && period.type == 'lecture' && <TText p={5} size={15} color={course.data.started ? "red" : "green"}>{(!course.data.started && "Start Course") || (course.data.started && "Stop Course")}</TText>}
-        </TView >
+
+            {user?.data.type === 'student' && course?.data.started && period.type === 'lecture' && (
+                <TText p={5} size={15} color="red">Join Course</TText>
+            )}
+
+            {user?.data.type === 'professor' && period.type === 'lecture' && (
+                <TText p={5} size={15} color={course.data.started ? "red" : "green"}>
+                    {course.data.started ? "Stop Course" : "Start Course"}
+                </TText>
+            )}
+        </TView>
```

Comment:
Firstly I want say that it very clear what you are trying to achieve with this code. I like the way you handle the `"Start course"` `"Stop course"`.
I just have a small pist of improvement. Let start fron the fact that a professor might not be in a quantum superposition state of changing everymillisecond from `professor` to `student`. Therefore, as we know directly on the start what is the starte of the user we could leverage this piece of information and modularize a bit the code.
Maybe consider creating a function that return the associated component. In example:

```typescript
const renderStudentAction = (course: { id: string; data: Course }, period: CourseTimePeriod) => {
    if (course?.data.started && period.type === 'lecture') {
        return <TText p={5} size={15} color="red">Join Course</TText>;
    }
    return null;
};


const renderProfessorAction = (course: { id: string; data: Course }, period: CourseTimePeriod) => {
    if (period.type === 'lecture') {
        return (
            <TText p={5} size={15} color={course.data.started ? "red" : "green"}>
                {course.data.started ? "Stop Course" : "Start Course"}
            </TText>
        );
    }
    return null;
};
```

and then in the assignement phse before the `return` you could directly create a `const toBeRendered = //depending on the status`. Better than this, put it in a usefeccct so that if somehow the professor turns into a superposition state, we could also handle this case 🤗.
Hope you see my point.